src/optimizer/cardinality_estimator.cpp

```diff
@@ -0,0 +1,379 @@
+//
+// Created by Tom Ebergen on 7/6/22.
+//
+#include "duckdb/planner/filter/conjunction_filter.hpp"
+#include "duckdb/planner/filter/constant_filter.hpp"
+#include "duckdb/optimizer/join_order_optimizer.hpp"
+#include "duckdb/planner/operator/logical_get.hpp"
+#include "duckdb/storage/data_table.hpp"
+#include "duckdb/storage/statistics/numeric_statistics.hpp"
+
+namespace duckdb {
+
+static TableCatalogEntry *GetCatalogTableEntry(LogicalOperator *op) {
+	if (op->type == LogicalOperatorType::LOGICAL_GET) {
+		auto get = (LogicalGet *)op;
+		TableCatalogEntry *entry = get->GetTable();
+		return entry;
+	}
+	for (auto &child : op->children) {
+		TableCatalogEntry *entry = GetCatalogTableEntry(child.get());
+		if (entry != nullptr) {
+			return entry;
+		}
+	}
+	return nullptr;
+}
+
+bool CardinalityEstimator::SingleColumnFilter(FilterInfo *filter_info) {
+	if (filter_info->left_set && filter_info->right_set) {
+		// Both set
+		return false;
+	}
+	//! Filter on one relation, (i.e string or range filter on a column).
```

Comment:
For these comments we use just `//`, not `//!`.

In general, `//!` is reserved for describing a function/field, for example in a header file.